test/jdk/java/net/httpclient/http2/server/OutgoingPushPromise.java

```diff
@@ -33,6 +36,7 @@ class OutgoingPushPromise extends Http2Frame {
     final URI uri;
     final InputStream is;
     final int parentStream; // not the pushed streamid
+    private LinkedList<ContinuationFrame> continuations;
```

Comment:
Make this `final` and instantiate it in the constructor instead. 
Could use `List<ContinuationFrame>` here - or even `List<Http2Frame>` if you want to simulate the bug we had before - e.g - you could add a `HeaderFrame` instead of a `ContinuationFrame`, and verify the client no longer hangs and rightfully closes the connection...